audience_manager/api.rst

```diff
@@ -0,0 +1,213 @@
+additionally requires 2 changes in example code.
+
+App ID
+``````
+Replace 'XXX-XXX-XXX-XXX-XXX' with your Piwik App ID (previously Website ID). Example::
+
+    "efcd98a5-335b-48b0-ab17-bf43f1c542be"
+
+PPMS domain
+```````````
+Replace 'ppms.example.com' with your PPMS tracker domain. Please note that this value is used in 2 places:
+
+* 'https://ppms.example.com/audiences/static/widget/audience-manager.api.min.js'
+* 'ppms.example.com'
+
+Snippet
+```````
+
+Code:
+
+.. code-block:: html
+
+    <script>
+        (function(a,d,g,h,b,c,e){a[b]=a[b]||{};a[b][c]=a[b][c]||{};a[b][c][e]=a[b][c][e]||function(){(a[b][c][e].q=a[b][c][e].q||[]).push(arguments)};var f=d.createElement(g);d=d.getElementsByTagName(g)[0];f.async=1;f.src=h;d.parentNode.insertBefore(f,d)})
+        (window,document,"script","https://ppms.example.com/audiences/static/widget/audience-manager.api.min.js","ppms","am","api");
+
+        ppms.am.api('create', 'XXX-XXX-XXX-XXX-XXX', 'ppms.example.com');
+    </script>
+
+API function
+------------
+
+Loading snippet creates following API function:
+
+.. function:: ppms.am.api(command, ...args)
+
+    JavaScript API interface.
+
+    :param string command: Command name.
+    :param args: Command arguments. Number of arguments and their function depend on command.
+    :returns: Nothing. All commands are prepared to be run asynchronously.
+
+
+Commands
+--------
+Following commands work in context of current browser user profile. Additionally these commands require communication
+with server so they are asynchronous. Callback functions are used to provide response or error information.
+
+List audiences
+``````````````
+Gets list of user audiences.
+
+Code::
+
+    ppms.am.api('getAudiences', onFulfilled, onRejected);
+
+.. function:: onFulfilled(audience_list)
+
+    Function executed on success.
+
+    :param Array<string> audience_list: Array of audience IDs user belongs to. Example::
+
+        ['e8c6e873-955c-4771-9fd5-92c94577e9d9', '756e5920-422f-4d13-b73a-917f696ca288']
+
+.. function:: onRejected(error)
+
+    Function executed on error.
+
+    :param string error: Error code. Example::
+
+        'server_error'
+
+Check audience
+``````````````
+Checks if user is part of specific audience.
+
+Code::
+
+    ppms.am.api('checkAudience', audience_id, onFulfilled, onRejected);
+
+.. data:: audience_id
+
+    ID of checked audience. Example::
+
+        '52073260-5861-4a56-be5e-6628794722ee'
+
+.. function:: onFulfilled(in_audience)
+
+    Function executed on success.
+
+    :param boolean in_audience: True when user is part of audience, false otherwise.
+
+.. function:: onRejected(error)
+
+    Function executed on error.
+
+    :param string error: Error code. Example::
+
+        'server_error'
+
+Get attributes
+``````````````
+Gets user attributes.
+
+.. warning::
+
+    Only white-listed attributes are available this way.
```

Comment:
Where can I manage the whitelist? Why are only whitelisted attributes available? Take a look at backend API docs for an inspiration: http://am.devtools.piwik.pro/#tag/attributes%2Fpaths%2F~1audiences~1public~1api~1profiles~1%7Banalytics_visitor_id%7D~1attributes~1%2Fget